### PR TITLE
CO-3595 add pagination and change/move some call to improve performance

### PR DESCRIPTION
**Pull request by JonathanGuerne**
- fix internal error on donations when user doesn't have any sponsorship
- add pagination
- improve loading performance by moving function call to backend

### CODE REVIEW

File: website_compassion/controllers/my_account.py [hunk 1]
```diff
@@ -270,24 +272,43 @@ def my_child(self, state="active", child_id=None, **kwargs):
...
             ("type", "=", "out_invoice"),
             ("amount_total", "!=", 0),
-        ])
+        ]
+
+        sudo_invoice_env = request.env["account.invoice"].sudo()
+
+        all_invoice_count = sudo_invoice_env.search_count(invoice_search_criteria)
+
+        invoice_per_page = 30
```
Comment (coup1234) [inline website_compassion/controllers/my_account.py]: Suggestion: maybe the invoice_per_page can be a parameter of the function with default = 30

File: website_compassion/controllers/my_account.py [hunk 2]
```diff
@@ -296,23 +317,25 @@ def my_donations(self, form_id=None, **kw):
             ("date_invoice", "<", fields.Date.to_string(in_one_month))
         ])
 
+        last_completed_tax_receipt = partner.last_completed_tax_receipt
```
Comment (coup1234) [inline website_compassion/controllers/my_account.py:323]: Maybe this init is not really needed as we only use it on time (line 406)

Comment (JonathanGuerne) [reply website_compassion/controllers/my_account.py:323]: Yes the call is actually only needed in the creation of the dictionary used in the return statement. But when called directly in the dictionary it led to a strange behavior where last_completed_tax year was computed not only for the current user but also for other people. Maybe it has something to do with the state of self changing during the dictionary preparation. either way calling it before, as it is done here, solved the issue

**Review by ecino — APPROVED** [review]

File: website_compassion/controllers/my_account.py [hunk 3]
```diff
@@ -356,27 +381,31 @@ def my_donations(self, form_id=None, **kw):
...
             ("partner_id", "=", partner.id),
             ("state", "=", "paid"),
             ("type", "=", "out_invoice"),
             ("amount_total", "!=", 0),
-        ], limit=1, order="create_date asc")
+        ], limit=1, order="create_date asc").create_date
+
         current_year = datetime.today().year
-        first_year = first_invoice.create_date and first_invoice.create_date.year or current_year
+        first_year = create_date.year if create_date or current_year
```
Comment (ecino) [inline website_compassion/controllers/my_account.py]:
```suggestion
        first_year = create_date.year if create_date else current_year
```

**Pull request merged**